crypto/rand/drbg_lib.c

```diff
@@ -394,9 +489,23 @@ int RAND_DRBG_generate(RAND_DRBG *drbg, unsigned char *out, size_t outlen,
         reseed_required = 1;
     }
 
-    if (drbg->reseed_counter >= drbg->reseed_interval)
+    if (drbg->generate_counter >= drbg->reseed_interval)
         reseed_required = 1;
 
+    if (drbg->reseed_time > 0) {
+        /* automatic reseeding is enabled */
+        if (drbg->parent != NULL
+            && drbg->reseed_counter != drbg->parent->reseed_counter)
+            reseed_required = 1;
+        else {
+            time_t now = time(NULL);
+
+            if (now < drbg->reseed_time
+                || now - drbg->reseed_time >= drbg->reseed_time_interval)
+                reseed_required = 1;
+        }
```

Comment:
I think the whole block should not be in the else, both the parent and child should check the reseed_time_interval.

Reply:
At least based on other comment in the file, that's not really what we want. We want to check that the master's reseed_time_interval is up at some place, and then make sure a generate call is made. This doesn't seem to be happening anywhere currently. One way of doing that is check the parent here and if needed set reseed_required.

Reply:
Oh, but our reseed_time equals the parent's reseed_time, so removing the else is what we wanted anyway.

Reply:
Which makes me wonder why we copy this, while we could just look at it? And that really doesn't seem to work when there are 3 DRBGs in the chain?

Reply:
I guess I see 2 options:
- Actually look at the parent's variables. This will probably require taking a lock which we want to avoid here.
- Split the variables, so there is a set for the current drbg and the parent drbg

Reply:
Having a copy doesn't work for the RAND_add() case

Reply:
> And that really doesn't seem to work when there are 3 DRBGs in the chain?

You are right, this does not work, but it was never intended to. See my review summary.